### PR TITLE
DSN JSON to tscircuit JSON: complete smt pads parsing

### PR DESCRIPTION
**Pull request by andrii-balitskyi**
- **Refactor parsing pad name**
- **Adjust to the update pad parsing, convert round rect to rect**
- **Update test snapshot**

### CODE REVIEW

File: pcb-json-to-tscircuit-soup-json/index.ts [hunk 1]
```diff
@@ -65,42 +60,80 @@ export const convertDsnJsonToTscircuitSoupJson = (
...
-      const pad = {
+      const padShapeMap = {
+        Round: "circle",
+        Oval: "circle",
+        RoundRect: "rect",
+        Rect: "rect",
+        Cust: "rect",
+      }
+      const padLayerMap = {
+        A: "top",
```
Comment (seveibar) [inline pcb-json-to-tscircuit-soup-json/index.ts]: This is what it's supposed to be, there can be no `inner` layers or up to 20 depending on the number of available layers

```suggestion
        A: ["top", "inner1", "inner2", "bottom"],
```

File: pcb-json-to-tscircuit-soup-json/parse-pad-name.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+export function parsePadName(padName: string): PadInfo {
+  const parts = padName.split("_")
+  const [shapeAndLayer, ...rest] = parts
+
+  if (!shapeAndLayer) {
+    throw new Error(`Unable to parse pad shape and layer from ${padName}`)
+  }
+
+  const shapeMatch = shapeAndLayer.match(/([A-Za-z]+)(?:\[([ABT])\])?/)
```
Comment (seveibar) [inline pcb-json-to-tscircuit-soup-json/parse-pad-name.ts:19]: do we still need this file? I thought we didn't need to parse the pad names anymore

File: tests/simpledsn1/__snapshots__/basic-dsn-snapshot.snap.svg [hunk 3]
```diff
@@ -1,11 +1,11 @@
 <svg xmlns="http://www.w3.org/2000/svg" width="800" height="600"><style>
```
Comment (seveibar) [inline tests/simpledsn1/__snapshots__/basic-dsn-snapshot.snap.svg:1]: something got messed up in this snapshot

Comment (seveibar) [reply tests/simpledsn1/__snapshots__/basic-dsn-snapshot.snap.svg:1]: if you look at the image on github, it seems like there's probably a really crazy coordinate or something, i would just look at the soup and see if you can avoid regressing the snapshot

File: pcb-json-to-tscircuit-soup-json/parse-pad-name.ts (hunk 2 above)
Comment (andrii-balitskyi) [reply pcb-json-to-tscircuit-soup-json/parse-pad-name.ts:19]: I think we still need to parse width and height of an oval pad to transform it into a circle and calculate the diameter.
Here's how oval pad definition usually looks like:
```
    (padstack Oval[A]Pad_2286x1524_um
      (shape (path F.Cu 1524  -381 0  381 0))
      (shape (path B.Cu 1524  -381 0  381 0))
      (attach off)
    )
```
The shape usually defines path width (1524) - thickness of the oval's "stroke" and path endpoints (-381 and 381) - the horizontal extent of the oval.

File: tests/simpledsn1/__snapshots__/basic-dsn-snapshot.snap.svg (hunk 3 above)
Comment (andrii-balitskyi) [reply tests/simpledsn1/__snapshots__/basic-dsn-snapshot.snap.svg:1]: I wonder if `mm` lib is not working correctly here

Comment (andrii-balitskyi) [reply tests/simpledsn1/__snapshots__/basic-dsn-snapshot.snap.svg:1]: I wonder if `mm` lib is not working as expected

Comment (andrii-balitskyi) [reply tests/simpledsn1/__snapshots__/basic-dsn-snapshot.snap.svg:1]: Yea, when I divide values by 1000, I get different results

**Review by seveibar — APPROVED** [review]

**Pull request closed without merging**